internal/overlord/servstate/handlers.go

```diff
@@ -162,11 +177,23 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 			}
 			releasePlan()
 		}
-		return fmt.Errorf("cannot start service: exited quickly with code %d", cmd.ProcessState.ExitCode())
+		msg := fmt.Sprintf("cannot start service: exited quickly with code %d", cmd.ProcessState.ExitCode())
```

Comment:
we should be able to read the last log here and omit the `lastLog` variable

Reply:
I actually had that at first, but there's a race condition between the goroutine that calls `cmd.Wait()`, because it's closing the log buffer. So if the goroutine's `logBuffer.Close()` occurs before or during this reads the last logs, we get an error reading the logs because the log buffer is already closed.

Reply:
Per discussion, this was a bug in the ring buffer where it would double-close `iterator.closeChan` if you created the iterator after the ring buffer `Close()` method was called (which should only close it for writes, not reads). Fixed the bug per your suggestion and updated the test to catch this (it wasn't closing the iterators).